bgpd/bgp_attr.c

```diff
@@ -2681,6 +2656,40 @@ bgp_attr_parse_ret_t bgp_attr_parse(struct peer *peer, struct attr *attr,
 		return BGP_ATTR_PARSE_ERROR;
 	}
 
+	/*
+	 * RFC4271: If the NEXT_HOP attribute field is syntactically incorrect,
+	 * then the Error Subcode MUST be set to Invalid NEXT_HOP Attribute.
+	 * This is implemented below and will result in a NOTIFICATION. If the
+	 * NEXT_HOP attribute is semantically incorrect, the error SHOULD be
+	 * logged, and the route SHOULD be ignored. In this case, a NOTIFICATION
+	 * message SHOULD NOT be sent. This is implemented elsewhere.
+	 *
+	 * RFC4760: An UPDATE message that carries no NLRI, other than the one
+	 * encoded in the MP_REACH_NLRI attribute, SHOULD NOT carry the NEXT_HOP
+	 * attribute. If such a message contains the NEXT_HOP attribute, the BGP
+	 * speaker that receives the message SHOULD ignore this attribute.
+	 */
+	if (CHECK_FLAG(attr->flag, ATTR_FLAG_BIT(BGP_ATTR_NEXT_HOP))
```

Comment:
Nikos, 

The code changes looks good. One question though. Is there any specific reason to replace bgp_attr_malformed directly with bgp_notify_send() ? In BGP_ATTR_NEXTHOP case, eventually bgp_attr_malformed() function will call bgp_notify_send(), but I am not sure if they does exactly the same thing. 

Let me know, I will approve this accordingly.

Reply:
You are correct that eventually bgp_notify_send will be called but it's not needed to go through all the checks and code in bgp_attr_malformed or check if we should relax sending a notification, hence why I made the change.